airflow/providers_manager.py

```diff
@@ -61,6 +62,36 @@ def _create_customized_form_field_behaviours_schema_validator():
     return validator
 
 
+def _sanity_check(provider_package: str, class_name: str) -> bool:
+    """
+    Performs sanity check on provider classes.
+    For apache-airflow providers - it checks if it starts with appropriate package. For all providers
+    it tries to import the provider - checking that there are no exceptions during importing.
+    """
+    if provider_package.startswith("apache-airflow"):
+        provider_path = provider_package[len("apache-") :].replace("-", ".")
+        if not class_name.startswith(provider_path):
+            log.warning(
+                "Sanity check failed when importing '%s' from '%s' package. It should start with '%s'",
+                class_name,
+                provider_package,
+                provider_path,
+            )
+            return False
+    try:
+        module, class_name = class_name.rsplit('.', maxsplit=1)
+        getattr(importlib.import_module(module), class_name)
```

Comment:
Can we use [`import_string`](https://github.com/apache/airflow/blob/main/airflow/utils/module_loading.py#L22) here?

Reply:
Ah . I did not even know we had it . Sure :)